aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/S3Event.java

```diff
@@ -53,4 +57,22 @@ private S3Event(String event) {
     public String toString() {
         return this.event;
     }
+    
+	private static final Map<String, S3Event> stringToEnum = new HashMap<>();
+
+	static {
+		for(S3Event s3Event : S3Event.values()) {
+			stringToEnum.put(s3Event.toString().toUpperCase(), s3Event);
+		}
+	}
+
+	public static S3Event fromString(String key) {
```

Comment:
Our other `enum` use `fromValue`, can we use the same name here?

Let's also follow the same basic logic used in the generated enums: https://github.com/aws/aws-sdk-java/blob/0549ecc7b625dbe1be44169a874408e241c8d1d8/aws-java-sdk-api-gateway/src/main/java/com/amazonaws/services/apigateway/model/SecurityPolicy.java#L37-L59

Reply:
Thanks for the feedback, @dagnir. I changed this in the new commit, following the same logic suggested. I kept the extra bit where if the `value` doesn't match, I give it another chance by prefixing it with `s3:`. This was because the events receiving from S3 are like this:
`"eventName": "ObjectRestore:Completed"`